src/middlewared/middlewared/plugins/cluster_linux/ctdb_private.py

```diff
@@ -0,0 +1,241 @@
+from middlewared.schema import Dict, IPAddr
+from middlewared.service import (accepts, private, job,
+                                 Service, ValidationErrors,
+                                 CallError)
+from middlewared.plugins.cluster_linux.utils import CTDBConfig
+import pathlib
+
+
+SHARED_VOL = CTDBConfig.CTDB_LOCAL_MOUNT.value
+PRI_IP_FILE = CTDBConfig.GM_PRI_IP_FILE.value
+ETC_IP_FILE = CTDBConfig.ETC_PRI_IP_FILE.value
+PRI_LOCK = CTDBConfig.PRI_LOCK.value
+
+
+class CtdbIpService(Service):
+
+    class Config:
+        namespace = 'ctdb.private.ips'
+
+    @private
+    def query(self):
+
+        ips = []
+
+        # logic is as follows:
+        #   1. if ctdb daemon is started
+        #       ctdb just reads the `ETC_IP_FILE` and loads the
+        #       ips written there into the cluster. However,
+        #       if a private ip is added/removed, it doesn't
+        #       mean the ctdb cluster has been reloaded to
+        #       see the changes in the file. So return what
+        #       the daemon sees.
+        #   2. else
+        #       if the `SHARED_VOL` is mounted and `ETC_IP_FILE` exists
+        #       and is a symlink and the symlink is pointed to the
+        #       `PRI_IP_FILE` then read it and return the contents
+        if self.middleware.call_sync('service.started', 'ctdb'):
+            ips = [i['address'] for i in self.middleware.call_sync('ctdb.general.listnodes')]
+        else:
+            try:
+                mounted = pathlib.Path(SHARED_VOL).is_mount()
+            except Exception:
+                # can happen when mounted but glusterd service
+                # is stopped/crashed etc
+                mounted = False
+
+            if mounted:
+                pri_ip_file = pathlib.Path(PRI_IP_FILE)
+                etc_ip_file = pathlib.Path(ETC_IP_FILE)
+                if pri_ip_file.exists():
+                    if etc_ip_file.is_symlink() and etc_ip_file.resolve() == pri_ip_file:
+                        with open(PRI_IP_FILE) as f:
+                            ips = f.read().splitlines()
+        return ips
```

Comment:
To expand on William's comment, you can include `filter_list` from middlewared.utils and then do something like:
```
return filter_list(ips, filters, options)
```